### PR TITLE
AEA-1155 removal of default stub connection

### LINKED ISSUE #1967: Remove stub connection as default connection
**Is your feature request related to a problem? Please describe.**
Currently the default, it creates an unnecessary overhead

**Describe the solution you'd like**
Remove, no default connection is needed.

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR removes the stub connection as a default. It makes the necessary changes and also allows for AEAs to be run without a connection at all.

## Fixes

#1967 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/cli/interact.py [hunk 1]
```diff
@@ -63,6 +62,8 @@
     )
     from packages.fetchai.protocols.default.message import DefaultMessage  # noqa: F401
 
+STUB_CONNECTION = "fetchai/stub:latest"
```
Comment (DavidMinarsch) [inline aea/cli/interact.py:65]: Move to cli constants

File: tests/test_cli/test_utils/test_utils.py [hunk 2]
```diff
@@ -472,7 +471,7 @@ def test_is_item_present_unified(mock_, vendor):
         (PublicId.from_str("author/package:latest"), False),
         (PublicId.from_str("fetchai/oef:0.1.0"), False),
         (PublicId.from_str("fetchai/oef:latest"), False),
-        (PublicId.from_str(DEFAULT_CONNECTION), False),
+        (PublicId.from_str("fetchai/stub:lastest"), False),
```
Comment (DavidMinarsch) [inline tests/test_cli/test_utils/test_utils.py:474]: Fix typo

**Review by solarw — APPROVED** [review]
LGTM

**Review by marcofavorito — APPROVED** [review]
LGTM

**Pull request merged**